src/ast/passes/codegen_llvm.cpp

```diff
@@ -3807,6 +3820,50 @@ void CodegenLLVM::createIncDec(Unop &unop)
   }
 }
 
+Function *CodegenLLVM::createMapSizeCallback()
+{
+  // The goal is to produce the following code:
+  //
+  // static int cb(struct map *map, void *key, void *value, void *ctx)
+  // {
+  //   int *size = (int *)ctx;
+  //   (*size)++;
+  //   return 0;
+  // }
+  auto saved_ip = b_.saveIP();
+
+  std::array<llvm::Type *, 4> args = {
+    b_.getInt8PtrTy(), b_.getInt8PtrTy(), b_.getInt8PtrTy(), b_.getInt8PtrTy()
+  };
+
+  FunctionType *callback_type = FunctionType::get(b_.getInt64Ty(), args, false);
+
+  Function *callback = Function::Create(callback_type,
+                                        Function::LinkageTypes::InternalLinkage,
+                                        "map_size_cb",
+                                        module_.get());
+
+  callback->addFnAttr(Attribute::NoInline);
```

Comment:
Just curious, why is this needed?

Reply:
Good question, I copied it from the original PR. I believe that the original point is to prevent Clang from inlining the function, although I'd hope that it never does that since it's used as a callback. Still, it's probably no harm leaving this attribute.

Reply:
> I would assume this as well. Is it easy to find out if this is the case or are there different clang build environments/settings that might cause this to happen in the future?

Considering how much LLVM likes to break things, I'm usually mostly concerned about future versions. But you're right that this shouldn't happen, kernel selftests also don't use the `noline` keyword. Removed.